bugsnag-android-core/src/androidTest/java/com/bugsnag/android/BugsnagTestUtils.java

```diff
@@ -132,4 +136,83 @@ public static AppWithState generateAppWithState() {
     public static App generateApp() {
         return new App(generateImmutableConfig(), null, null, null, null, null);
     }
+
+    /**
+     * "Normalize" a map by changing all numeric types to their largest forms.
+     * This is necessary for comparing the results of serialization/deserialization
+     * operations because we have no control over what types the codec will choose,
```

Comment:
Will the output not be predictable with the codec? I feel I'm missing something over why these methods are necessary

Reply:
When deserializing, I encountered issues where it would parse a number as a long, and in other cases as an int. This would break naive equality checks.